### PR TITLE
doc: io_trace feature to debug IO perf

### PR DESCRIPTION
**Pull request by jakmeier**
A deep dive into using IO traces to understand IO performance.

### CODE REVIEW

**Comment by jakmeier** [comment]
@pugachAG when you have a moment (this is obviously not urgent) please read through these new docs regarding IO traces.

Please let me know if anything is unclear. At the absolute minimum, a proficient storage engineer at Pagoda should be able to make sense of these docs, or else I didn't explain things well. ;)

**Comment by jakmeier** [comment]
@pugachAG just noticed that this docs PR is still hanging, did you have a change to look at it?

cc @jbajic

File: docs/practices/workflows/io_trace.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+The output contains one `apply_transactions` for each chunk, with the block hash
+and the shard id. Then it prints one line for each DB operations observed
+(GET,SET,...) together with a list of columns and an OP count.
+
+See the `top-level` output at the end? These are all the DB requests that could
+not be assigned to specific chunks. The way we currently count write operations
+(SET, UPDATE_RC) they are never assigned to a specific chunk and instead only
+show up in the top-level list. Clearly, there is some room for improvement here.
+So far we simply did not worried about RocksDB write performance so the tooling
```
Comment (pugachAG) [inline docs/practices/workflows/io_trace.md:323]: nit: "did not worry" or "haven't worried"

**Review by pugachAG — APPROVED** [review]
This is awesome, thanks @jakmeier for taking time to write such a detailed doc!

**Pull request merged**